projects/graph/src/graph.py

```diff
@@ -1,9 +1,168 @@
 """
 Simple graph implementation
 """
+"""
+{
+'0': {'1', '3'},
+'1': {'0'},
+'2': {set()},
+'3': {'0'}
+}
+"""
+import pprint
+from queue import *
+
+
+class Vertex:
+    def __init__(self, name, neighbors=list(), distance=9999, color='black'):
+        self.name = name #label
+        self.neighbor = list() #same as edges?
+        self.distance = 9999 #destination
+        self.color = 'black' #black for visited, red for not visited
+
 
 
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+        self.edges = set()
+        # self.destination = destination
+        # self.label = label
+
+
+
+    def add_vertex(self, vertex):
+        self.vertices[vertex] = set()
+        print(self.vertices)
+
+    def add_edge(self, vertex_start, vertex_end):
+        if vertex_start not in self.vertices:
+            print(f'You chose: {vertex_start}. That Beginning Vertex does not exist...')
+            return
+        if vertex_end not in self.vertices:
+            print(f"You chose: {vertex_end} That End Vertex does not exist...")
+            return
+        for i in self.vertices:
+            # temp = set()
+            # temp.add(i)
+            if i == vertex_start:
+                self.vertices[vertex_start].add(vertex_end)
+                print(self.vertices)
+            if vertex_end == i:
+                self.vertices[vertex_end].add(vertex_start)
+                print(self.vertices)
+
+
+
+    def breadth_first_traverse(self, starting_vertex):
+        # create a _queue_ FIFO
+        q = []
+        visited = set()
+        # Enqueue the starting vertex
+        q.append(starting_vertex)
+        print(q)
+        # while the queue is not empty
+        while len(q) > 0:
+            # dequeue a node from the queue
+            deq = q.pop()
+
+            # Mark it as visited
+            visited.add(deq)
+            print("visited: ",visited)
+            # Enqueue all of its children
+            for i in self.vertices[deq]:
+                if i not in visited:
+                    q.append(i)
+                if len(q) == 0:
+                    return visited
+
+
+    def depth_first_traverse(self, starting_vertex):
+        # create a stack LIFO
+        s = queue.LifoQueue()
```

Comment:
You can create your own Queue class in another file or at the top of this file here. This might be easier than importing from the built-in module.